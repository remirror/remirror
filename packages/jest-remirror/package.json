{
  "name": "jest-remirror",
  "version": "3.0.0-beta.6",
  "description": "A text editor for react built with prosemirror",
  "keywords": [
    "prosemirror",
    "react",
    "text editor"
  ],
  "homepage": "https://github.com/remirror/remirror/tree/HEAD/packages/jest-remirror",
  "repository": {
    "type": "git",
    "url": "https://github.com/remirror/remirror.git",
    "directory": "packages/jest-remirror"
  },
  "license": "MIT",
  "contributors": [
    "Ifiok Jr. <ifiokotung@gmail.com>"
  ],
  "type": "module",
  "exports": {
    ".": {
      "types": "./dist-types/index.d.ts",
      "import": "./dist/jest-remirror.js",
      "require": "./dist/jest-remirror.cjs"
    },
    "./environment": {
      "types": "./dist-types/environment.d.ts",
      "import": "./environment/dist/jest-remirror-environment.js",
      "require": "./environment/dist/jest-remirror-environment.cjs"
    },
    "./package.json": "./package.json"
  },
  "main": "./dist/jest-remirror.cjs",
  "module": "./dist/jest-remirror.js",
  "types": "./dist-types/index.d.ts",
  "files": [
    "dist",
    "dist-types",
    "environment"
  ],
  "scripts": {
    "build": "remirror-cli build"
  },
  "dependencies": {
    "@babel/runtime": "^7.22.3",
    "@remirror/core": "3.0.0-beta.6",
    "@remirror/dom": "3.0.0-beta.6",
    "@remirror/pm": "3.0.0-beta.4",
    "@remirror/preset-core": "3.0.0-beta.6",
    "@testing-library/dom": "^9.3.0",
    "@types/sanitize-html": "^2.9.0",
    "jest-prosemirror": "3.0.0-beta.4",
    "sanitize-html": "^2.10.0",
    "tiny-warning": "^1.0.3"
  },
  "devDependencies": {
    "@remirror/cli": "1.1.0-beta.0",
    "@types/node": "^16.18.34",
    "jest": "^29.5.0"
  },
  "peerDependencies": {
    "@types/node": "*",
    "jest": ">= 24.9.0"
  },
  "peerDependenciesMeta": {
    "@types/node": {
      "optional": true
    },
    "jest": {
      "optional": true
    }
  },
  "publishConfig": {
    "access": "public",
    "exports": {
      ".": {
        "types": "./dist/jest-remirror.d.ts",
        "import": "./dist/jest-remirror.js",
        "require": "./dist/jest-remirror.cjs"
      },
      "./environment": {
        "types": "./environment/dist/jest-remirror-environment.d.ts",
        "import": "./environment/dist/jest-remirror-environment.js",
        "require": "./environment/dist/jest-remirror-environment.cjs"
      },
      "./package.json": "./package.json"
    },
    "main": "./dist/jest-remirror.cjs",
    "module": "./dist/jest-remirror.js",
    "type": "module",
    "types": "./dist/jest-remirror.d.ts"
  },
  "@remirror": {
    "tsconfigs": {
      "src": {
        "compilerOptions": {
          "types": [
            "node",
            "jest"
          ]
        }
      }
    }
  },
  "preconstruct": {
    "entrypoints": [
      "index.ts",
      "environment.ts"
    ]
  }
}
