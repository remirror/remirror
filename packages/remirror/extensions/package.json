{
  "type": "module",
  "exports": {
    ".": {
      "types": "./dist/remirror-extensions.d.ts",
      "import": "./dist/remirror-extensions.js",
      "require": "./dist/remirror-extensions.cjs"
    },
    "./package.json": "./package.json"
  },
  "main": "./dist/remirror-extensions.cjs",
  "module": "./dist/remirror-extensions.js",
  "types": "./dist/remirror-extensions.d.ts"
}
