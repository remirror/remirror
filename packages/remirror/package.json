{
  "name": "remirror",
  "version": "3.0.0-beta.8",
  "description": "One editing package to rule them all, one editing package to bind them.",
  "keywords": [
    "prosemirror",
    "react",
    "remirror",
    "wysiwyg",
    "editing",
    "text editor"
  ],
  "homepage": "https://github.com/remirror/remirror/tree/HEAD/packages/remirror",
  "repository": {
    "type": "git",
    "url": "https://github.com/remirror/remirror.git",
    "directory": "packages/remirror"
  },
  "license": "MIT",
  "contributors": [
    "Ifiok Jr. <ifiokotung@gmail.com>",
    "ocavue "
  ],
  "sideEffects": false,
  "type": "module",
  "exports": {
    ".": {
      "types": "./dist-types/index.d.ts",
      "import": "./dist/remirror.js",
      "require": "./dist/remirror.cjs"
    },
    "./dom": {
      "types": "./dist-types/dom.d.ts",
      "import": "./dom/dist/remirror-dom.js",
      "require": "./dom/dist/remirror-dom.cjs"
    },
    "./extensions": {
      "types": "./dist-types/extensions.d.ts",
      "import": "./extensions/dist/remirror-extensions.js",
      "require": "./extensions/dist/remirror-extensions.cjs"
    },
    "./icons": {
      "types": "./dist-types/icons.d.ts",
      "import": "./icons/dist/remirror-icons.js",
      "require": "./icons/dist/remirror-icons.cjs"
    },
    "./package.json": "./package.json",
    "./styles/*": "./styles/*"
  },
  "main": "./dist/remirror.cjs",
  "module": "./dist/remirror.js",
  "types": "./dist-types/index.d.ts",
  "files": [
    "dist",
    "dist-types",
    "dom",
    "extensions",
    "icons",
    "styles"
  ],
  "scripts": {
    "build": "remirror-cli build"
  },
  "dependencies": {
    "@babel/runtime": "^7.22.3",
    "@remirror/core": "3.0.0-beta.8",
    "@remirror/core-constants": "3.0.0-beta.4",
    "@remirror/core-helpers": "4.0.0-beta.5",
    "@remirror/core-types": "3.0.0-beta.5",
    "@remirror/core-utils": "3.0.0-beta.6",
    "@remirror/dom": "3.0.0-beta.8",
    "@remirror/extension-annotation": "3.0.0-beta.8",
    "@remirror/extension-bidi": "3.0.0-beta.8",
    "@remirror/extension-blockquote": "3.0.0-beta.8",
    "@remirror/extension-bold": "3.0.0-beta.8",
    "@remirror/extension-callout": "3.0.0-beta.8",
    "@remirror/extension-code": "3.0.0-beta.8",
    "@remirror/extension-code-block": "3.0.0-beta.8",
    "@remirror/extension-collaboration": "3.0.0-beta.8",
    "@remirror/extension-columns": "3.0.0-beta.8",
    "@remirror/extension-diff": "3.0.0-beta.8",
    "@remirror/extension-doc": "3.0.0-beta.8",
    "@remirror/extension-drop-cursor": "3.0.0-beta.8",
    "@remirror/extension-embed": "3.0.0-beta.8",
    "@remirror/extension-emoji": "3.0.0-beta.8",
    "@remirror/extension-entity-reference": "3.0.0-beta.8",
    "@remirror/extension-epic-mode": "3.0.0-beta.8",
    "@remirror/extension-events": "3.0.0-beta.8",
    "@remirror/extension-find": "1.0.0-beta.8",
    "@remirror/extension-font-family": "3.0.0-beta.8",
    "@remirror/extension-font-size": "3.0.0-beta.8",
    "@remirror/extension-gap-cursor": "3.0.0-beta.8",
    "@remirror/extension-hard-break": "3.0.0-beta.8",
    "@remirror/extension-heading": "3.0.0-beta.8",
    "@remirror/extension-history": "3.0.0-beta.8",
    "@remirror/extension-horizontal-rule": "3.0.0-beta.8",
    "@remirror/extension-image": "3.0.0-beta.8",
    "@remirror/extension-italic": "3.0.0-beta.8",
    "@remirror/extension-link": "3.0.0-beta.8",
    "@remirror/extension-list": "3.0.0-beta.8",
    "@remirror/extension-markdown": "3.0.0-beta.8",
    "@remirror/extension-mention": "3.0.0-beta.8",
    "@remirror/extension-mention-atom": "3.0.0-beta.8",
    "@remirror/extension-node-formatting": "3.0.0-beta.8",
    "@remirror/extension-paragraph": "3.0.0-beta.8",
    "@remirror/extension-placeholder": "3.0.0-beta.8",
    "@remirror/extension-positioner": "3.0.0-beta.8",
    "@remirror/extension-shortcuts": "3.0.0-beta.8",
    "@remirror/extension-strike": "3.0.0-beta.8",
    "@remirror/extension-sub": "3.0.0-beta.8",
    "@remirror/extension-sup": "3.0.0-beta.8",
    "@remirror/extension-tables": "3.0.0-beta.8",
    "@remirror/extension-text": "3.0.0-beta.8",
    "@remirror/extension-text-case": "3.0.0-beta.8",
    "@remirror/extension-text-color": "3.0.0-beta.8",
    "@remirror/extension-text-highlight": "3.0.0-beta.8",
    "@remirror/extension-trailing-node": "3.0.0-beta.8",
    "@remirror/extension-underline": "3.0.0-beta.8",
    "@remirror/extension-whitespace": "3.0.0-beta.8",
    "@remirror/icons": "3.0.0-beta.5",
    "@remirror/preset-core": "3.0.0-beta.8",
    "@remirror/preset-formatting": "3.0.0-beta.8",
    "@remirror/preset-wysiwyg": "3.0.0-beta.8",
    "@remirror/theme": "3.0.0-beta.5",
    "@types/refractor": "^3.0.2",
    "refractor": "^3.6.0"
  },
  "devDependencies": {
    "@remirror/cli": "1.1.0-beta.1",
    "@remirror/pm": "3.0.0-beta.6",
    "@types/prettier": "^2.7.2",
    "prettier": "^2.8.8"
  },
  "peerDependencies": {
    "@remirror/pm": "^3.0.0-beta.6",
    "@types/prettier": "^2.7.2",
    "prettier": "^2.8.8"
  },
  "peerDependenciesMeta": {
    "@types/prettier": {
      "optional": true
    },
    "prettier": {
      "optional": true
    }
  },
  "publishConfig": {
    "access": "public",
    "exports": {
      ".": {
        "types": "./dist/remirror.d.ts",
        "import": "./dist/remirror.js",
        "require": "./dist/remirror.cjs"
      },
      "./dom": {
        "types": "./dom/dist/remirror-dom.d.ts",
        "import": "./dom/dist/remirror-dom.js",
        "require": "./dom/dist/remirror-dom.cjs"
      },
      "./extensions": {
        "types": "./extensions/dist/remirror-extensions.d.ts",
        "import": "./extensions/dist/remirror-extensions.js",
        "require": "./extensions/dist/remirror-extensions.cjs"
      },
      "./icons": {
        "types": "./icons/dist/remirror-icons.d.ts",
        "import": "./icons/dist/remirror-icons.js",
        "require": "./icons/dist/remirror-icons.cjs"
      },
      "./package.json": "./package.json",
      "./styles/*": "./styles/*"
    },
    "main": "./dist/remirror.cjs",
    "module": "./dist/remirror.js",
    "type": "module",
    "types": "./dist/remirror.d.ts"
  },
  "@remirror": {},
  "preconstruct": {
    "entrypoints": [
      "index.ts",
      "dom.ts",
      "extensions.ts",
      "icons.ts"
    ]
  }
}
