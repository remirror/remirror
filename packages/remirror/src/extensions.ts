// Extensions
export * from '@remirror/extension-annotation';
export * from '@remirror/extension-bidi';
export * from '@remirror/extension-blockquote';
export * from '@remirror/extension-bold';
export * from '@remirror/extension-callout';
export * from '@remirror/extension-code';
export * from '@remirror/extension-code-block';
export * from '@remirror/extension-codemirror5';
export * from '@remirror/extension-collaboration';
export * from '@remirror/extension-columns';
export * from '@remirror/extension-diff';
export * from '@remirror/extension-doc';
export * from '@remirror/extension-drop-cursor';
export * from '@remirror/extension-embed';
export * from '@remirror/extension-emoji';
export * from '@remirror/extension-entity-reference';
export * from '@remirror/extension-epic-mode';
export * from '@remirror/extension-events';
export * from '@remirror/extension-find';
export * from '@remirror/extension-font-family';
export * from '@remirror/extension-font-size';
export * from '@remirror/extension-gap-cursor';
export * from '@remirror/extension-hard-break';
export * from '@remirror/extension-heading';
export * from '@remirror/extension-history';
export * from '@remirror/extension-horizontal-rule';
export * from '@remirror/extension-image';
export * from '@remirror/extension-italic';
export * from '@remirror/extension-link';
export * from '@remirror/extension-list';
export * from '@remirror/extension-markdown';
export * from '@remirror/extension-mention';
export * from '@remirror/extension-mention-atom';
export * from '@remirror/extension-node-formatting';
export * from '@remirror/extension-paragraph';
export * from '@remirror/extension-placeholder';
export * from '@remirror/extension-positioner';
export * from '@remirror/extension-shortcuts';
export * from '@remirror/extension-strike';
export * from '@remirror/extension-sub';
export * from '@remirror/extension-sup';
export * from '@remirror/extension-tables';
export * from '@remirror/extension-text';
export * from '@remirror/extension-text-case';
export * from '@remirror/extension-text-color';
export * from '@remirror/extension-text-highlight';
export * from '@remirror/extension-trailing-node';
export * from '@remirror/extension-underline';
export * from '@remirror/extension-whitespace';
export * from '@remirror/extension-yjs';

// Presets
export * from '@remirror/preset-core';
export * from '@remirror/preset-formatting';
export * from '@remirror/preset-wysiwyg';
