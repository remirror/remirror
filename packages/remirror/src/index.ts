export * from '@remirror/core';
export * from '@remirror/theme';
