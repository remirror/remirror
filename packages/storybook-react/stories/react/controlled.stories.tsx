import { useCallback, useState } from 'react';
import { Remirror, ThemeProvider, useRemirror, useRemirrorContext } from '@remirror/react';

const DOC = {
  type: 'doc',
  content: [
    {
      type: 'paragraph',
      content: [
        {
          type: 'text',
          text: 'New content',
        },
      ],
    },
  ],
};

export default { title: 'Editors / Controlled' };

const extensions = () => [];

export const Editable = (): JSX.Element => {
  const { manager, state, onChange } = useRemirror({
    extensions,
    content: '<p>Some text</p>',
    stringHandler: 'html',
  });

  const [editable, setEditable] = useState(true);

  return (
    <ThemeProvider>
      <Remirror manager={manager} state={state} onChange={onChange} editable={editable} />
      <button onClick={() => setEditable(!editable)}>
        Toggle editable (currently {editable.toString()})
      </button>
    </ThemeProvider>
  );
};

function SetContentButton() {
  const { setContent } = useRemirrorContext();
  return <button onClick={() => setContent(DOC)}>Replace content</button>;
}

export const ReplaceContentPreserveHistory = (): JSX.Element => {
  const { manager, state, onChange } = useRemirror({
    extensions,
    content: '<p>Original content</p>',
    stringHandler: 'html',
  });

  return (
    <ThemeProvider>
      <Remirror manager={manager} state={state} onChange={onChange} autoRender='end'>
        <SetContentButton />
      </Remirror>
    </ThemeProvider>
  );
};

export const ReplaceContentAndClearHistory = (): JSX.Element => {
  const { manager, state, onChange } = useRemirror({
    extensions,
    content: '<p>Original content</p>',
    stringHandler: 'html',
  });

  const handleClick = useCallback(() => {
    // Clear out old state when setting data from outside
    // This prevents e.g. the user from using CTRL-Z to go back to the old state
    manager.view.updateState(manager.createState({ content: DOC }));
  }, [manager]);

  return (
    <ThemeProvider>
      <Remirror manager={manager} state={state} onChange={onChange} />
      <button onClick={handleClick}>Replace content</button>
    </ThemeProvider>
  );
};
