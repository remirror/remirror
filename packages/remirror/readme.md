# remirror

> One package to rule them all, one entry point to bind them.

[![Version][version]][npm] [![Weekly Downloads][downloads-badge]][npm] [![Bundled size][size-badge]][size] [![Typed Codebase][typescript]](#) [![MIT License][license]](#)

[version]: https://flat.badgen.net/npm/v/remirror
[npm]: https://npmjs.com/package/remirror
[license]: https://flat.badgen.net/badge/license/MIT/purple
[size]: https://bundlephobia.com/result?p=remirror
[size-badge]: https://flat.badgen.net/bundlephobia/minzip/remirror
[typescript]: https://flat.badgen.net/badge/icon/TypeScript?icon=typescript&label
[downloads-badge]: https://badgen.net/npm/dw/remirror/red?icon=npm

## Installation

```bash
# yarn
yarn add remirror @remirror/pm

# pnpm
pnpm add remirror @remirror/pm

# npm
npm install remirror @remirror/pm
```

The `remirror` package will automatically install the `@remirror/pm` package for you. You only need to install it yourself if you'd like to ensure consistent versions of the `prosemirror-*` libraries when importing from `@remirror/pm/state` instead of `prosemirror-state` or `@remirror/pm/model` instead of `prosemirror-model`.

## Usage

Rather than installing multiple scoped packages, the `remirror` package is a gateway to using all the goodness that remirror provides while minimising your bundle size.

The following creates a controlled editor with React.

```tsx
import React from 'react';
import { SocialPreset } from 'remirror/extensions';
import { Remirror, SocialEmojiComponent, useRemirror } from '@remirror/react';

const EditorWrapper = () => {
  const socialPreset = new SocialPreset();
  const { state, onChange } = useRemirror({ extensions: () => [...socialPreset()] });

  return (
    <Remirror state={state} onChange={onChange} manager={manager} autoRender={true}>
      <SocialEmojiComponent />
    </Remirror>
  );
};
```

These are the entry points available through the `remirror` package.

- `remirror` - All the core functionality available through `@remirror/core`.
- `remirror/extensions` - All the core extensions and presets made available through the main `remirror` repository. This doesn't include any framework specific extensions and presets.
- `remirror/dom` - The dom framework implementation of via `createDomEditor`.
