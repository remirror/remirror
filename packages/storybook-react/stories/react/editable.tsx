import { useState } from 'react';
import { Remirror, ThemeProvider, useRemirror } from '@remirror/react';

const Editable = (): JSX.Element => {
  const { manager, state, onChange } = useRemirror({
    extensions: () => [],
    content: '<p>Some text</p>',
    stringHandler: 'html',
  });

  const [editable, setEditable] = useState(true);

  return (
    <ThemeProvider>
      <Remirror manager={manager} state={state} onChange={onChange} editable={editable} />
      <button onClick={() => setEditable(!editable)}>
        Toggle editable (currently {editable.toString()})
      </button>
    </ThemeProvider>
  );
};

export default Editable;
