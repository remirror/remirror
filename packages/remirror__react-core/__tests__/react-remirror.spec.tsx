import { jest } from '@jest/globals';
import { RemirrorTestChain } from 'jest-remirror';
import React, { FC } from 'react';
import { BoldExtension, ItalicExtension } from 'remirror/extensions';
import { docNodeBasicJSON } from 'testing';
import { fireEvent, strictRender } from 'testing/react';
import type { I18nFormatter, RemirrorMessage } from '@remirror/core';

import { createReactManager, Remirror, useI18n, useRemirrorContext } from '../';

test('`Remirror`', () => {
  const TestComponent: FC = () => {
    const { getRootProps } = useRemirrorContext();
    return (
      <div>
        <div data-testid='target' {...getRootProps()} />
      </div>
    );
  };

  const manager = createReactManager(() => []);

  const { getByRole, getByTestId } = strictRender(
    <Remirror initialContent={docNodeBasicJSON} manager={manager}>
      <TestComponent />
    </Remirror>,
  );
  const target = getByTestId('target');
  const editor = getByRole('textbox');

  expect(target).toContainElement(editor);
});

test('multiple `getRootProps` applied to dom throw an error', () => {
  const spy = jest.spyOn(console, 'error').mockImplementation(() => {});
  const manager = createReactManager(() => []);

  const TestComponent: FC = () => {
    const { getRootProps } = useRemirrorContext();

    return (
      <div>
        <div data-testid='target' {...getRootProps()} />
        <div data-testid='target1' {...getRootProps()} />
      </div>
    );
  };

  expect(() =>
    strictRender(
      <Remirror manager={manager}>
        <TestComponent />
      </Remirror>,
    ),
  ).toThrowErrorMatchingSnapshot();

  spy.mockRestore();
});

test('can run multiple transaction based commands', () => {
  const TestComponent: FC = () => {
    const { getRootProps, commands } = useRemirrorContext<BoldExtension | ItalicExtension>();

    return (
      <div>
        <div data-testid='target' {...getRootProps()} />
        <button
          onClick={() => {
            commands.toggleBold();
            commands.toggleItalic();
          }}
        />
      </div>
    );
  };

  const manager = createReactManager([new BoldExtension(), new ItalicExtension()]);
  const chain = RemirrorTestChain.create(manager);

  const { getByRole } = strictRender(
    <Remirror initialContent={docNodeBasicJSON} manager={manager}>
      <TestComponent />
    </Remirror>,
  );

  const { p, doc } = chain.nodes;
  const { bold, italic } = chain.marks;

  chain.add(doc(p('<start>This<end>')));
  fireEvent.click(getByRole('button'));

  expect(chain.state.doc).toEqualRemirrorDocument(doc(p(bold(italic('This')))));
});

test('can run multiple synchronous non chainable commands', () => {
  const TestComponent: FC = () => {
    const { getRootProps, view } = useRemirrorContext<BoldExtension | ItalicExtension>();

    return (
      <div>
        <div data-testid='target' {...getRootProps()} />
        <button
          onClick={() => {
            view.dispatch(view.state.tr.insertText('a'));
            view.dispatch(view.state.tr.insertText('b'));
          }}
        />
      </div>
    );
  };

  const manager = createReactManager([]);
  const chain = RemirrorTestChain.create(manager);

  const { getByRole } = strictRender(
    <Remirror initialContent={docNodeBasicJSON} manager={manager}>
      <TestComponent />
    </Remirror>,
  );

  const { p, doc } = chain.nodes;

  chain.add(doc(p('')));
  fireEvent.click(getByRole('button'));

  expect(chain.state.doc).toEqualRemirrorDocument(doc(p('ab')));
});

describe('`autoRender`', () => {
  it('renders at the start', () => {
    const manager = createReactManager(() => []);
    const { container } = strictRender(
      <Remirror autoRender='start' manager={manager}>
        <div>Hello</div>
      </Remirror>,
    );

    expect(container.innerHTML).toMatchSnapshot();
  });

  it('renders at the start: when set to true', () => {
    const manager = createReactManager(() => []);
    const { container } = strictRender(
      <Remirror autoRender={true} manager={manager}>
        <div>Hello</div>
      </Remirror>,
    );

    expect(container.innerHTML).toMatchSnapshot();
  });

  it('renders at the end', () => {
    const manager = createReactManager(() => []);
    const { container } = strictRender(
      <Remirror autoRender='end' manager={manager}>
        <div>Hello</div>
      </Remirror>,
    );

    expect(container.innerHTML).toMatchSnapshot();
  });

  it('renders by default with no children', () => {
    const manager = createReactManager(() => []);
    const { container } = strictRender(<Remirror manager={manager} />);

    expect(container.innerHTML).toMatchSnapshot();
  });
});

describe('i18nFormat', () => {
  const message: RemirrorMessage = {
    id: 'some.test.id',
    message: 'Some ICU formatted message to be parsed with {blank}',
    comment: 'This is a message for unit test purposes',
  };

  const values = {
    blank: 'your chosen i18n library',
  };

  const TestComponent: FC = () => {
    const t = useI18n();
    return <button>{t(message, values)}</button>;
  };

  it('exposes the passed i18nFormat function on the useI18n hook', () => {
    const i18nFormat = jest.fn<I18nFormatter>().mockImplementation(({ id }) => id);
    const manager = createReactManager(() => []);
    strictRender(
      <Remirror manager={manager} i18nFormat={i18nFormat}>
        <TestComponent />
      </Remirror>,
    );

    expect(i18nFormat).toHaveBeenCalledWith(message, values, 'en', ['en']);
  });

  it('exposes the passed locale information on the useI18n hook', () => {
    const i18nFormat = jest.fn<I18nFormatter>().mockImplementation(({ id }) => id);
    const manager = createReactManager(() => []);
    strictRender(
      <Remirror
        manager={manager}
        i18nFormat={i18nFormat}
        locale='fr'
        supportedLocales={['fr', 'de']}
      >
        <TestComponent />
      </Remirror>,
    );

    expect(i18nFormat).toHaveBeenCalledWith(message, values, 'fr', ['fr', 'de']);
  });

  it('useI18n returns the value returned by the i18nFormat function', () => {
    const i18nFormat = jest.fn<I18nFormatter>().mockImplementation((msg, values = {}) => {
      let { message } = msg;

      for (const [key, value] of Object.entries(values)) {
        message = message.replaceAll(`{${key}}`, value);
      }

      return message;
    });
    const manager = createReactManager(() => []);
    const { getByRole } = strictRender(
      <Remirror manager={manager} i18nFormat={i18nFormat}>
        <TestComponent />
      </Remirror>,
    );

    expect(getByRole('button')).toMatchInlineSnapshot(`
      <button>
        Some ICU formatted message to be parsed with your chosen i18n library
      </button>
    `);
  });
});
