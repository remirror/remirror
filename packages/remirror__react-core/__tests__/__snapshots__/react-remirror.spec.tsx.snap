// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`autoRender\` renders at the end 1`] = `
<div>
  Hello
</div>
<div class="remirror-editor-wrapper">
  <div
    contenteditable="true"
    role="textbox"
    aria-multiline="true"
    aria-label
    aria-placeholder
    translate="no"
    class="ProseMirror remirror-editor"
  >
    <p
      data-placeholder
      class="remirror-is-empty"
    >
      <br class="ProseMirror-trailingBreak">
    </p>
  </div>
</div>
`;

exports[`\`autoRender\` renders at the start 1`] = `
<div class="remirror-editor-wrapper">
  <div
    contenteditable="true"
    role="textbox"
    aria-multiline="true"
    aria-label
    aria-placeholder
    translate="no"
    class="ProseMirror remirror-editor"
  >
    <p
      data-placeholder
      class="remirror-is-empty"
    >
      <br class="ProseMirror-trailingBreak">
    </p>
  </div>
</div>
<div>
  Hello
</div>
`;

exports[`\`autoRender\` renders at the start: when set to true 1`] = `
<div class="remirror-editor-wrapper">
  <div
    contenteditable="true"
    role="textbox"
    aria-multiline="true"
    aria-label
    aria-placeholder
    translate="no"
    class="ProseMirror remirror-editor"
  >
    <p
      data-placeholder
      class="remirror-is-empty"
    >
      <br class="ProseMirror-trailingBreak">
    </p>
  </div>
</div>
<div>
  Hello
</div>
`;

exports[`\`autoRender\` renders by default with no children 1`] = `
<div class="remirror-editor-wrapper">
  <div
    contenteditable="true"
    role="textbox"
    aria-multiline="true"
    aria-label
    aria-placeholder
    translate="no"
    class="ProseMirror remirror-editor"
  >
    <p
      data-placeholder
      class="remirror-is-empty"
    >
      <br class="ProseMirror-trailingBreak">
    </p>
  </div>
</div>
`;

exports[`multiple \`getRootProps\` applied to dom throw an error 1`] = `
"\`getRootProps\` has been attached to the DOM more than once. It should only be attached to the dom once per editor.

Called 2 times

For more information visit https://remirror.io/docs/errors#rmr0201"
`;
